File: Easy/1290. Convert Binary Number in a Linked List to Integer/getDecimalValue.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public int getDecimalValue(ListNode head) {
        String s = "";
        while(true){
            s += Integer.toString(head.val); 
            head = head.next;
            if(head == null){
                break;
            }
        }
        return Integer.valueOf(s,2); //轉10進位
    }
}

File: Easy/1290. Convert Binary Number in a Linked List to Integer/getDecimalValue.cpp
class Solution {
public:
    int getDecimalValue(ListNode* head) {
        /*方法1
        while(true){
            res += to_string(head->val);
            head = head->next;
            if(!head){
                break;
            }
        }
        
        return stoi(res,0,2);
        */
        //方法二
        int res = 0;
        while(head){
            res = res * 2 + head->val;
            head = head->next;
        }
        return res;
    }
};